h2o-core/src/main/java/hex/faulttolerance/Recovery.java

```diff
@@ -13,8 +19,30 @@
 import java.util.*;
 
 public class Recovery<T extends Keyed> {
+    
+    private static final Logger LOG = Logger.getLogger(Recovery.class);
 
     public static final String REFERENCES_META_FILE_SUFFIX = "_references";
+    public static final String RECOVERY_META_FILE = "recovery.json";
+    
+    public static final String INFO_CLASS = "class";
+    public static final String INFO_RESULT_KEY = "resultKey";
+    public static final String INFO_JOB_KEY = "jobKey";
+
+    public static void autoRecover(String autoRecoveryDir) {
+        if (autoRecoveryDir == null || autoRecoveryDir.length() == 0) {
```

Comment:
`Optional` and `Path`s or `File`s are much more type safe. The problem with Path and File throughout the code is serialization, let's skip that. But using Optional to indicate an argument is Optional seems like a good idea to me.

Reply:
done